src/test/java/com/example/cucumber/gradle/parallel/demo/StepGlue.java

```diff
@@ -3,31 +3,30 @@
 import io.cucumber.java.en.Then;
 import io.cucumber.java.en.When;
 import org.junit.jupiter.api.Assertions;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.test.web.client.TestRestTemplate;
-import org.springframework.boot.web.server.LocalServerPort;
 
 public class StepGlue {
-
-    @LocalServerPort
+    @Value("${server.port.forTest}")
     private int port;
 
-    @Autowired
-    @SuppressWarnings("SpringJavaAutowiredMembersInspection") // Handled by Cucumber
-    private TestRestTemplate testRestTemplate;
+    @Value("${app.title}")
+    private String title;
+
+    private TestRestTemplate testRestTemplate =  new TestRestTemplate();
 
-    private static String lastEcho;
+    private String lastEcho;
```

Comment:
This field here and all the references to it need to be `static`.
I know that for this example, it is useless, but it simulates the current behavior and constraints of our real (And much much more complex) app.

This being static ensures that any parallelization, either by DB or by Feature/Scenario needs to be run in a separate JVM if we want the test to pass.